gsd/hoomd.py

```diff
@@ -530,6 +531,19 @@ def __len__(self):
         """ The number of frames in the trajectory. """
         return self.file.nframes;
 
+    def __getstate__(self):
+        '''Checks if in read mode and removes file handle from __dict__'''
+        if self.file.mode not in ['rb', 'rb+']:
+            raise PickleError("Only trajectories in read only "
+                              "mode can be pickled.")
+        state = dict(name=self.file.name, mode=self.file.mode)
+        return state
+
+    def __setstate__(self, state):
+        '''Reopens file handle using gsd.hoomd.open'''
+        self.__dict__ = open(name=state['name'], mode=state['mode']).__dict__
```

Comment:
This always opens the file using the C file layer API. Should we support pickling of trajectories opened with the pure python reader in `pygsd.py`? Or should we raise an error in this case?

Reply:
I just decided to make the underlying `GSDFile` objects themselves pickable and this solves that problem.